packages/jupyter-ai/jupyter_ai/chat_handlers/base.py

```diff
@@ -289,6 +289,8 @@ def close_pending(self, pending_msg: PendingMessage):
             handler.broadcast_message(close_pending_msg)
             break
 
+        pending_msg.closed = True
```

Comment:
Can you add an early return at the top of this `close_pending()` method to make sure only one `ClosePendingMessage` gets sent to each pending message? This isn't an issue right now, but it seems reasonable that this method should be idempotent.

```py
if pending_msg.closed:
    return
```